bmwqemu.pm

```diff
@@ -305,6 +306,22 @@ sub save_json_file {
     return rename("$fn.new", $fn);
 }
 
+=head2 save_yaml_file
+
+  save_yaml_file([$data, $fn]);
+
+Store validate and save yaml string into the given filename.
+=cut
+sub save_yaml_file {
+    my ($data, $fn) = @_;
+    open(my $fd, ">", "$fn.new");
+    my $yp = YAML::PP->new;
+    my $yaml = $yp->load_string($data);
+    print $fd $yp->dump_string($yaml);
```

Comment:
I would actually suggest to use `dump_file`, as it is shorter and will do utf8 encoding automatically. Otherwise you should do the encoding yourself.
Also I would recommend to rename the variabes. `$data` is a YAML string, and `$yaml` is a native data structure. Readers will expect the opposite.
```
$yp->dump_file("$fn.new", $data);
```